libecs/include/pivot/ecs/Core/Scene.hxx

```diff
@@ -76,67 +82,19 @@ public:
 
     // System methods
 
-    /// Register a system before its usage
-    /// @code
-    /// gCoordinator.RegisterSystem<{YourSystem}>();
-    /// @endcode
-    template <typename T>
-    std::shared_ptr<T> RegisterSystem()
-    {
-        auto system = mSystemManager->RegisterSystem<T>();
-        mSystems.push_back(system);
-        return system;
-    }
-
-    /// Set a signature on your System to get the good list of entity.
-    /// In this exemple, the system will pass on all entity wich have {YourComponent}
-    /// @code
-    /// Signature signature;
-    /// signature.set(gCoordinator.GetComponentType<{YourComponent}>());
-    /// signature.set(gCoordinator.GetComponentType<Transform>());
-    /// gCoordinator.SetSystemSignature<{YourSystem}>(signature);
-    /// @endcode
-    template <typename T>
-    void SetSystemSignature(Signature signature)
-    {
-        mSystemManager->SetSignature<T>(signature);
-        for (Entity entity = 0; entity < getLivingEntityCount(); entity++) {
-            if ((getSignature(entity) & signature) == signature) mSystemManager->setEntityToSystem<T>(entity);
-        }
-    }
-
-    /// Check if a system is register in a scene
-    template <typename T>
-    bool hasSystem()
-    {
-        return mSystemManager->hasSystem<T>();
-    }
-
-    /// Update registered systems
-    void Update(float dt);
+    /// Get the system manager
+    pivot::ecs::systems::Manager &getSystemManager();
```

Comment:
for small one-liners, prefer implementing them in the header file has the compiler will be able to inline them